src/main/java/com/treasuredata/android/TreasureData.java

```diff
@@ -685,7 +688,12 @@ public void appendModelInformation(Map<String, Object> record) {
         record.put(EVENT_KEY_DEVICE, Build.DEVICE);
         record.put(EVENT_KEY_MODEL, Build.MODEL);
         record.put(EVENT_KEY_OS_VER, Build.VERSION.SDK_INT);
-        record.put(EVENT_KEY_OS_TYPE, OS_TYPE);
+        UiModeManager uiModeManager = (UiModeManager) applicationContext.getSystemService(UI_MODE_SERVICE);
+        if (uiModeManager.getCurrentModeType() == Configuration.UI_MODE_TYPE_TELEVISION) {
```

Comment:
Should we hoist this somewhere instead of recalculating for each event?

Reply:
Thanks! Hoisted in [6e2c3d8](https://github.com/treasure-data/td-android-sdk/pull/58/commits/6e2c3d8afadb2cb076f018eff9bbe71213a049fd)